Formula/sonarqube.rb

```diff
@@ -16,13 +16,26 @@ class Sonarqube < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux: "5bd7b5e142d30a1caf30979e5e077f462e076540c514505d354df4625e2be67d"
   end
 
-  # sonarqube ships pre-built x86_64 binaries
-  depends_on arch: :x86_64
+  depends_on "java-service-wrapper" => :build
```

Comment:
Why not make this a runtime dependency and make symlinks instead of copy files?

Reply:
This is possible for binaries and for the jar file, but the script needs to be copied and modified.

Do you think it is worth using a build dependency for the script and a run dependency for the links ?

Reply:
Yes. Copy the script but symlink everything everything else.

Reply:
The problem with this approach is that the script and the war file need to be at the same version. If the war file is symlinked and the script is copied, when the java-service-wrapper will be upgraded the war file will be upgraded but not the script. This means that upgrading a dependency will broke an installed formula.

In my opinion, it's safer to copy everything as now. What do you think ?

Reply:
When we update something, we test all parts that depend on it, so we would catch that in CI.

Reply:
I've done what you suggested:
```
-  depends_on "java-service-wrapper" => :build
+  depends_on "java-service-wrapper"
   depends_on "openjdk@11"

   conflicts_with "sonarqube-lts", because: "both install the same binaries"
@@ -24,9 +24,9 @@ class Sonarqube < Formula
   def install
     # Use Java Service Wrapper 3.5.46 which is Apple Silicon compatible
     jsw_libexec = Formula["java-service-wrapper"].opt_libexec
-    cp jsw_libexec/"lib/wrapper.jar", "#{buildpath}/lib/jsw/wrapper-3.5.46.jar"
-    cp jsw_libexec/"lib/libwrapper.dylib", "#{buildpath}/bin/macosx-universal-64/lib/"
-    cp jsw_libexec/"bin/wrapper", "#{buildpath}/bin/macosx-universal-64/"
+    ln_s jsw_libexec/"lib/wrapper.jar", "#{buildpath}/lib/jsw/wrapper-3.5.46.jar"
+    ln_s jsw_libexec/"lib/libwrapper.dylib", "#{buildpath}/bin/macosx-universal-64/lib/"
+    ln_sf jsw_libexec/"bin/wrapper", "#{buildpath}/bin/macosx-universal-64/"
     cp jsw_libexec/"scripts/App.sh.in", "#{buildpath}/bin/macosx-universal-64/sonar.sh"
     sonar_sh_file = "bin/macosx-universal-64/sonar.sh"
     inreplace sonar_sh_file, "@app.name@", "SonarQube"
```
But then sonarqube doesn't start anymore. If I replace the symbolic link to jsw_libexec/"bin/wrapper", by an hard link it works again. Should I use an hard link or copy the wrapper file ?

Reply:
Hard links wouldn't work with the precompiled binaries in homebrew. I guess copying is the only way.

Reply:
Ok, the last commit link the jar file and the library and copy the script and the wrapper binary.